### PR TITLE
#1341 update java client library feature flag methods

### PR DESCRIPTION
**Pull request by kawcl**

### CODE REVIEW

File: clientlibs/java/src/main/java/org/upgradeplatform/utils/Utils.java [hunk 1]
```diff
@@ -20,16 +19,16 @@ public class Utils
...
-	public static final String INITIALIZE_USER= "api/v5/init";
-	public static final String SET_GROUP_MEMBERSHIP= "api/v5/groupmembership";
-	public static final String SET_WORKING_GROUP= "api/v5/workinggroup";
-	public static final String GET_ALL_EXPERIMENTS= "api/v5/assign";
-	public static final String MARK_EXPERIMENT_POINT= "api/v5/mark";
-	public static final String GET_ALL_FEATURE_FLAGS= "api/v5/featureflag";
-	public static final String LOG_EVENT= "api/v5/log";
-	public static final String SET_ALT_USER_IDS= "api/v5/useraliases";
-	public static final String ADD_MATRIC= "api/v5/metric";
+	private static final String apiVersion = "api/v5";
```
Comment (danoswaltCL) [inline clientlibs/java/src/main/java/org/upgradeplatform/utils/Utils.java:22]: as discussed, will need to be `v6` once the v6 client controller pr is merged in.

Comment (kawcl) [reply clientlibs/java/src/main/java/org/upgradeplatform/utils/Utils.java]: changed to v6. also removed a redundant request object

**Comment by danoswaltCL** [comment]
i have `hasFeatureFlag` changes for TS lib coming, btw @kawcl @bcb37

File: clientlibs/java/src/main/java/org/upgradeplatform/responsebeans/InitializeUser.java [hunk 2]
```diff
@@ -4,24 +4,15 @@
 import java.util.Map;
 
 public class InitializeUser {
-    private String id;
```
Comment (bcb37) [inline clientlibs/java/src/main/java/org/upgradeplatform/responsebeans/InitializeUser.java:7]: The response object still has the `id` in it so this is failing when I run the request. It seems like maybe they used to be identical but now we may not be able to use the response type to define the request.

File: clientlibs/java/src/main/java/org/upgradeplatform/client/ExperimentClient.java [hunk 3]
```diff
@@ -92,20 +90,20 @@ public void close() {
 
 	// Initialize user with userId
 	public void init(final ResponseCallback<InitializeUser> callbacks) {
-		InitializeUser initUser = new InitializeUser(this.userId, null, null);
+		InitializeUser initUser = new InitializeUser(null, null);
 		initializeUser(initUser, callbacks);
```
Comment (bcb37) [inline clientlibs/java/src/main/java/org/upgradeplatform/client/ExperimentClient.java:94]: In the `initializeUser` method, the request object is created with the response type `InitializeUser`, which should have `id` on it (but no longer does). Is it possible we need a new request bean for init?

**Review by bcb37 — DISMISSED** [review]
Requests and responses differ in v6 for `ExperimentUser` (used by the two 'group' endpoints), and `InitializeUser` (used by /init). I've created new request beans for these that work locally.

**Review by danoswaltCL — APPROVED** [review]

**Pull request merged**